### PR TITLE
fix(source_map): windows support

### PR DESCRIPTION
**Pull request by suxin2017**
## Related issue (if exists)

https://github.com/web-infra-dev/rspack/issues/3223
### <samp>🤖 Generated by Copilot at ae1ac46</samp>

This pull request fixes a bug and improves the code quality in `rspack_plugin_devtool`, and removes an unnecessary flag from the test script in `rspack/package.json`. These changes enhance the compatibility and maintainability of the rspack toolchain.

<details open=true>
  <summary><h2>Walkthrough</h2></summary>

### <samp>🤖 Generated by Copilot at ae1ac46</samp>

*  Fix source map paths on Windows to use forward slashes instead of backslashes ([link](https://github.com/web-infra-dev/rspack/pull/3239/files?diff=unified&w=0#diff-a3ed34d20f0b15a073bdeaac181d27e0fe74d19b63f7a0eb6cc1868192ef6dd9L115-R122), [link](https://github.com/web-infra-dev/rspack/pull/3239/files?diff=unified&w=0#diff-a3ed34d20f0b15a073bdeaac181d27e0fe74d19b63f7a0eb6cc1868192ef6dd9L239-R251))
* Remove unnecessary NODE_OPTIONS flag for the test script in the rspack package.json file ([link](https://github.com/web-infra-dev/rspack/pull/3239/files?diff=unified&w=0#diff-142a5c1a681bf6859bd9e53bbfe87fc9dd2d05e8350b0fb50bfcd652a9e36cbbL13-R13))

</details>

### CODE REVIEW

File: crates/rspack_plugin_devtool/src/lib.rs [hunk 1]
```diff
@@ -112,11 +112,14 @@ impl Plugin for DevtoolPlugin {
           map.set_file(Some(filename.clone()));
           for source in map.sources_mut() {
             let uri = normalize_custom_filename(source);
-            let resource_path = if let Some(relative_path) = diff_paths(uri, &context) {
+            let mut resource_path = if let Some(relative_path) = diff_paths(uri, &context) {
               relative_path.to_string_lossy().to_string()
             } else {
               uri.to_owned()
             };
+            if cfg!(target_os = "windows") {
```
Comment (hardfist) [inline crates/rspack_plugin_devtool/src/lib.rs:120]: seems weird to me, how webpack handle this problem?

Comment (suxin2017) [reply crates/rspack_plugin_devtool/src/lib.rs:120]: const _contextify = (context, request) => {
	return request
		.split("!")
		.map(r => absoluteToRequest(context, r))
		.join("!");
webpack is in here

Comment (hardfist) [reply crates/rspack_plugin_devtool/src/lib.rs:120]: can you paste the link

Comment (suxin2017) [reply crates/rspack_plugin_devtool/src/lib.rs:120]: const WINDOWS_ABS_PATH_REGEXP = /^[a-zA-Z]:[\\/]/;
const SEGMENTS_SPLIT_REGEXP = /([|!])/;
const WINDOWS_PATH_SEPARATOR_REGEXP = /\\/g;
const absoluteToRequest = (context, maybeAbsolutePath) => {
	if (maybeAbsolutePath[0] === "/") {
		if (
			maybeAbsolutePath.length > 1 &&
			maybeAbsolutePath[maybeAbsolutePath.length - 1] === "/"
		) {
			// this 'path' is actually a regexp generated by dynamic requires.
			// Don't treat it as an absolute path.
			return maybeAbsolutePath;
		}

		const querySplitPos = maybeAbsolutePath.indexOf("?");
		let resource =
			querySplitPos === -1
				? maybeAbsolutePath
				: maybeAbsolutePath.slice(0, querySplitPos);
		resource = relativePathToRequest(path.posix.relative(context, resource));
		return querySplitPos === -1
			? resource
			: resource + maybeAbsolutePath.slice(querySplitPos);
	}

	if (WINDOWS_ABS_PATH_REGEXP.test(maybeAbsolutePath)) {
		const querySplitPos = maybeAbsolutePath.indexOf("?");
		let resource =
			querySplitPos === -1
				? maybeAbsolutePath
				: maybeAbsolutePath.slice(0, querySplitPos);
		resource = path.win32.relative(context, resource);
		if (!WINDOWS_ABS_PATH_REGEXP.test(resource)) {
			resource = relativePathToRequest(
				resource.replace(WINDOWS_PATH_SEPARATOR_REGEXP, "/")
			);
		}
		return querySplitPos === -1
			? resource
			: resource + maybeAbsolutePath.slice(querySplitPos);
	}

	// not an absolute path
	return maybeAbsolutePath;
};

Comment (suxin2017) [reply crates/rspack_plugin_devtool/src/lib.rs:120]: He did it and I see we have this method in the warehouse, let me see how to fight it with him

Comment (suxin2017) [reply crates/rspack_plugin_devtool/src/lib.rs]: let source_path = contextify(&context, &source);
            let uri = normalize_custom_filename(&source_path);
            let resource_path = if let Some(relative_path) = diff_paths(uri, &context) {
              relative_path.to_string_lossy().to_string()
            } else {
              uri.to_owned()
            };
I think this implementation should work

**Review by ahabhgk — APPROVED** [review]
Thanks!

**Pull request merged**